File: ClientApp/src/app/modules/company/models/company.ts
import { Category } from "../../category/models/category";
import { Tag } from "../../tag/models/tag";
import { CompanyTag } from "./company-tag";

export class Company {
    Id:number;
    Title:String;
    Description:String;
    Content:String;
    Category:Category;
    PicsUrl:String;
    Publish4Time:Date;
    CompanyTags:CompanyTag[];
    constructor(){
        this.CompanyTags = [];
        this.Id = 0;
        this.Title = '';
        this.Publish4Time = null;
        this.PicsUrl ='';
        
    }
    static cloneBase(base: Company): Company {
        const result = new Company();
        result.Title = base.Title;
        result.Description = base.Description;
        result.Content = base.Content;
        if (base.CompanyTags == null){
            return result;
        }
        base.CompanyTags.forEach( tag => {
          result.CompanyTags.push(tag);
        });
        return result;
      }
}

File: DAL/Entity/Company.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
namespace MyBlogApp.DAL.Entity
{
    public class Company
    {
        public int Id { get; set; }
        public DateTime PublishTime { get; set; }

        [MaxLength(100, ErrorMessage = "Title of the company must be less than 100 symbs")]
        [Required (ErrorMessage = "Company must contain title")]
        public String Title { get; set; }
        public String Description { get; set; }
        public String PicsUrl { get; set; }

        [MaxLength(1000, ErrorMessage = "Company content length must be less than 1000 symbs")]
        [Required (ErrorMessage = "Company must contain content")]
        public String Content { get; set; }
        public List<CompanyTag> CompanyTags { get; set; }

        [Required(ErrorMessage = "Company must be part of some category")]
        public Category Category { get; set; }
    }
}
